### PR TITLE
[Ibis] Add ibis.schedule op

### PR DESCRIPTION
**Pull request by mortbopet**
The `ibis.schedule` operation defines a block wherein a group of operations can be placed wherein a limited number of Ibis threads can execute at once. The operation is isolated from above and has no return types. References to external values should be performed through SSA dominance and memrefs.

### CODE REVIEW

File: include/circt/Dialect/Ibis/IbisOps.td [hunk 1]
```diff
@@ -189,6 +190,29 @@ def ReturnOp : IbisOp<"return", [
...
+  HasParent<"MethodOp">,
+  SingleBlock,
+  NoRegionArguments,
+  NoTerminator
+]> {
+  let summary = "Ibis scheduling constraint block";
+  let description = [{
+    The `ibis.schedule` operation defines a block wherein a group of operations
+    can be placed wherein a limited number of Ibis threads can execute at once.
+    The operation is isolated from above and has no return types. References
```
Comment (teqdruid) [inline include/circt/Dialect/Ibis/IbisOps.td:203]: I don't see the isolated from above trait... did you mean *not* isolated from above?

**Review by teqdruid — COMMENTED** [review]
I'll leave approval to @blakep-msft to ensure this is something he'll be able to use.

**Review by blakep-msft — APPROVED** [review]
LGTM

File: include/circt/Dialect/Ibis/IbisOps.td (hunk 1 above)
Comment (mortbopet) [reply include/circt/Dialect/Ibis/IbisOps.td:203]: I did mean **not** - woops.

**Pull request merged**